cost-model/src/cost_model.rs

```diff
@@ -154,6 +154,24 @@ impl CostModel {
     fn get_transaction_cost(
         transaction: &impl TransactionWithMeta,
         feature_set: &FeatureSet,
+    ) -> (u64, u64, u64) {
+        if feature_set.is_active(&feature_set::reserve_minimal_cus_for_builtin_instructions::id()) {
+            let data_bytes_cost = Self::get_instructions_data_cost(transaction);
+            let (programs_execution_cost, loaded_accounts_data_size_cost) =
+                Self::get_estimated_execution_cost(transaction, feature_set);
+            (
+                programs_execution_cost,
+                loaded_accounts_data_size_cost,
+                data_bytes_cost,
+            )
+        } else {
+            Self::get_transaction_cost_without_minimal_builtin_cus(transaction, feature_set)
+        }
```

Comment:
To me it makes more sense for us to get try sanitizing `ComputeBudgetLimits` in this fn, before we branch on `reserve_minimal_cus_for_builtin_instructions`. and just pass that into the inner fns.

As is we are handling the `Err` case for sanitizing compute budget limits in 2 separate places - more room for an unintended difference to pop up.

Reply:
Other consideration is: not touching before-feature-gate code makes it much easier and safer to backport, and to remove after feature activation.

Reply:
That's fair